File: Samples/Scripts/Spark/Scala/GettingStarted/Foreach.scala
//Foreach: For every internet address(1), get the HTTP reply code(4) and print it as keyvalue pair from input 'NASA_Access_Log'
val input = sc.textFile("/Data/NASA_Access_Log")
	
val regex = "- ".r
	
val result = input.map(line => regex.replaceAllIn(line,""))
	
val pairs = result.map(line => (line.split(" (?=([^\"]*\"[^\"]*\")*[^\"]*$)+(?![^\\[]*\\])")(1),line.split(" (?=([^\"]*\"[^\"]*\")*[^\"]*$)+(?![^\\[]*\\])")(3)))

pairs.collect().foreach(println)

File: Samples/Scripts/Spark/Python/GettingStarted/Foreach.py
from pyspark import SparkContext
import re 

# Read the input file 
lines = sc.textFile("/Data/NASA_Access_Log")

# Method to split the input line and returns the word
def getKey(line, index):
	# Replace '- ' with empty value, so input lines are separated only by white space
	str = line.replace("- "," ")
	# Replace multiple spaces by single space
	str = ' '.join(str.split())
	# Split the input line by white space unless text enclosed with in double quotes and '[]' and stores the each field as string array
	str = re.findall('\[[^\]]*\]|\"[^\"]*\"|\S+', str)
	return str[index]

# Creates the Key-Value pair with the Internet Address as the key and the corresponding HTTP code as the value
pairs = lines.map(lambda s: (getKey(s,1), getKey(s,3)))
# Saves the created Key-Value pair in text file
pairs.saveAsTextFile("/Data/Output/ForeachPython"); 